src/StaticAnalyzers.cmake

```diff
@@ -38,11 +38,21 @@ macro(enable_cppcheck CPPCHECK_OPTIONS)
     set(CMAKE_C_CPPCHECK ${CMAKE_CXX_CPPCHECK})
 
     if(NOT "${CMAKE_CXX_STANDARD}" STREQUAL "")
-      set(CMAKE_CXX_CPPCHECK ${CMAKE_CXX_CPPCHECK} --std=c++${CMAKE_CXX_STANDARD})
+      if(CMAKE_CXX_STANDARD MATCHES [[03|11|14|17|20]])
+        set(CMAKE_CXX_CPPCHECK ${CMAKE_CXX_CPPCHECK} --std=c++${CMAKE_CXX_STANDARD})
+      else()
```

Comment:
What if cppcheck adds support for C++23?

Reply:
> Versions are hard-coded in the code, as cppcheck runs normally even if I run cppcheck --std=c++100 so I can't test whether the option is correct by running it.

That's a good reason.

Reply:
That makes sense. I'll give it another try in a few days.

Reply:
I think the current implementation is good enough for now. Detecting the supported standard is hard on our side. Cppcheck should probably provide some flags

Reply:
I think so too. Perhaps I can manually maintain version issues afterwards.